### ISSUE TITLE
PrestashopFacebook

### ISSUE THREAD

**Issue by Art3c** [PrestaShop/PrestaShop#24489]
After upgrading the module to vesion 1.9.0 the FO has stoped working. If the module is disable the FO is working OK. The BO is working in both cases normal.

Greatings, Tilen

Server conf:
MySQL: 10.3.28-MariaDB-log-cll-lve 
PHP: 7.3.27 
PrestaShop-a: 1.7.7.4 

PS Facebook Module

When debug is turned ON this is showing on the FO.

```
ServiceNotFoundException                                                                  The  service  "PrestaShop\Module\PrestashopFacebook\Factory\PsApiClientFactory" has a  dependency on a non-existent service  "PrestaShop\PsAccountsInstaller\Installer\Facade\PsAccounts".

in CheckExceptionOnInvalidReferenceBehaviorPass.php line 31
at CheckExceptionOnInvalidReferenceBehaviorPass->processValue(object(Reference), false)in AbstractRecursivePass.php line 60
at AbstractRecursivePass->processValue(array(object(Reference), object(Reference), object(Reference)), false)in CheckExceptionOnInvalidReferenceBehaviorPass.php line 28
at CheckExceptionOnInvalidReferenceBehaviorPass->processValue(array(object(Reference), object(Reference), object(Reference)))in AbstractRecursivePass.php line 67
at AbstractRecursivePass->processValue(object(Definition), true)in CheckExceptionOnInvalidReferenceBehaviorPass.php line 28
at CheckExceptionOnInvalidReferenceBehaviorPass->processValue(object(Definition), true)in AbstractRecursivePass.php line 60
at AbstractRecursivePass->processValue(array('service_container' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Adapter\ConfigurationAdapter' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Adapter\ToolsAdapter' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\API\FacebookCategoryClient' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\API\FacebookClient' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\API\EventSubscriber\AccountSuspendedSubscriber' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Buffer\TemplateBuffer' => object(Definition), 'ps_facebook.cache' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Config\Env' => object(Definition), 'ps_facebook.context' => object(Definition), 'ps_facebook.controller' => object(Definition), 'ps_facebook.cookie' => object(Definition), 'ps_facebook.language' => object(Definition), 'ps_facebook.currency' => object(Definition), 'ps_facebook.smarty' => object(Definition), 'ps_facebook.shop' => object(Definition), 'ps_facebook.link' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Dispatcher\EventDispatcher' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Factory\GoogleCategoryEssentialsApiClientFactory' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Factory\FacebookEssentialsApiClientFactory' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Factory\PsApiClientFactory' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\ApiConversionHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\CategoryMatchHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\ConfigurationHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\MessengerHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\PixelHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\ErrorHandler\ErrorHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\EventBusProductHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\PrevalidationScanRefreshHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Presenter\ModuleUpgradePresenter' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\AccessTokenProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\GoogleCategoryProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\FacebookDataProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\FbeDataProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\FbeFeatureDataProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\EventDataProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\ProductAvailabilityProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\MultishopDataProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\ProductSyncReportProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\PrevalidationScanCacheProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\PrevalidationScanDataProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Repository\GoogleCategoryRepository' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Repository\ProductRepository' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Repository\ShopRepository' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Repository\TabRepository' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Repository\ServerInformationRepository' => object(Definition), 'PrestaShop\Module\Ps_facebook\Translations\PsFacebookTranslations' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Manager\FbeFeatureManager' => object(Definition), 'PrestaShop\Module\Ps_facebook\Tracker\Segment' => object(Definition), 'ps_facebook' => object(Definition)), true)in CheckExceptionOnInvalidReferenceBehaviorPass.php line 28
at CheckExceptionOnInvalidReferenceBehaviorPass->processValue(array('service_container' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Adapter\ConfigurationAdapter' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Adapter\ToolsAdapter' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\API\FacebookCategoryClient' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\API\FacebookClient' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\API\EventSubscriber\AccountSuspendedSubscriber' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Buffer\TemplateBuffer' => object(Definition), 'ps_facebook.cache' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Config\Env' => object(Definition), 'ps_facebook.context' => object(Definition), 'ps_facebook.controller' => object(Definition), 'ps_facebook.cookie' => object(Definition), 'ps_facebook.language' => object(Definition), 'ps_facebook.currency' => object(Definition), 'ps_facebook.smarty' => object(Definition), 'ps_facebook.shop' => object(Definition), 'ps_facebook.link' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Dispatcher\EventDispatcher' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Factory\GoogleCategoryEssentialsApiClientFactory' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Factory\FacebookEssentialsApiClientFactory' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Factory\PsApiClientFactory' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\ApiConversionHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\CategoryMatchHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\ConfigurationHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\MessengerHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\PixelHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\ErrorHandler\ErrorHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\EventBusProductHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Handler\PrevalidationScanRefreshHandler' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Presenter\ModuleUpgradePresenter' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\AccessTokenProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\GoogleCategoryProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\FacebookDataProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\FbeDataProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\FbeFeatureDataProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\EventDataProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\ProductAvailabilityProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\MultishopDataProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\ProductSyncReportProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\PrevalidationScanCacheProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Provider\PrevalidationScanDataProvider' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Repository\GoogleCategoryRepository' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Repository\ProductRepository' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Repository\ShopRepository' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Repository\TabRepository' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Repository\ServerInformationRepository' => object(Definition), 'PrestaShop\Module\Ps_facebook\Translations\PsFacebookTranslations' => object(Definition), 'PrestaShop\Module\PrestashopFacebook\Manager\FbeFeatureManager' => object(Definition), 'PrestaShop\Module\Ps_facebook\Tracker\Segment' => object(Definition), 'ps_facebook' => object(Definition)), true)in AbstractRecursivePass.php line 39
at AbstractRecursivePass->process(object(ContainerBuilder))in Compiler.php line 140
at Compiler->compile(object(ContainerBuilder))in ContainerBuilder.php line 789
at ContainerBuilder->compile()in ContainerProvider.php line 91
at ContainerProvider->get('front')in ServiceContainer.php line 79
at ServiceContainer->initContainer()in ServiceContainer.php line 61
at ServiceContainer->getService('PrestaShop\\Module\\PrestashopFacebook\\Buffer\\TemplateBuffer')in ps_facebook.php line 200
at Ps_facebook->getService('PrestaShop\\Module\\PrestashopFacebook\\Buffer\\TemplateBuffer')in ps_facebook.php line 179
at Ps_facebook->__construct()
at ReflectionClass->newInstance()in Container.php line 123
at Container->makeInstanceFromClassName('ps_facebook', array('ps_facebook' => true))in Container.php line 153
at Container->doMake('ps_facebook', array('ps_facebook' => true))in Container.php line 166
at Container->make('ps_facebook')in ServiceLocator.php line 65
at ServiceLocator::get('ps_facebook')in Module.php line 1170
at ModuleCore::coreLoadModule('ps_facebook')in Module.php line 1149
at ModuleCore::getInstanceByName('ps_facebook')in Hook.php line 904
at HookCore::exec('actionFrontControllerSetMedia', array('cookie' => object(Cookie), 'cart' => object(Cart), 'altern' => 2))in FrontController.php line 949
at FrontControllerCore->setMedia()in Controller.php line 291
at ControllerCore->run()in Dispatcher.php line 518
at DispatcherCore->dispatch()in index.php line 28
```

**Comment by abcreationIT** [comment]
Me too.

PHP: 7.3.28
PrestaShop: 1.7.7.4

**Comment by florine2623** [comment]
Hello @Art3c @abcreationIT ,

We use GitHub issues only to discuss about bugs and new features in the PrestaShop project. If you have questions about using PrestaShop or third-party modules (You can contact PS Facebook Support directly support-facebook@prestashop.com), or if you need help with your shop, please consider one of our [support plans](https://www.prestashop.com/support).

Alternatively, you can also ask for help in the [community forums](https://www.prestashop.com/forums/) or in the [public Slack channel](https://github.com/PrestaShop/open-source/tree/master/slack).

Thank you

**Comment by Quetzacoalt91** [comment]
Hello @Art3c,

We released the version 1.9.1 of the module Facebook fixing this error, please upgrade to get rid of this issue.
